### PR TITLE
Add tests for OpenID STS creds and add to CI

### PR DESCRIPTION
**Pull request by donatello**
## Description

Adds openid IDP tests for STS API.

## Motivation and Context

Improve automated testing.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests and CI

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [x] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: .github/workflows/iam-integrations.yaml [hunk 1]
```diff
@@ -12,43 +12,20 @@ concurrency:
...
-  ldap-test:
-    name: LDAP Tests with Go ${{ matrix.go-version }}
+  iam-matrix-test:
+    name: "[Go=${{ matrix.go-version }}|ldap=${{ matrix.ldap }}|etcd=${{ matrix.etcd }}|openid=${{ matrix.openid }}]"
     runs-on: ubuntu-latest
 
     services:
       openldap:
-        image: quay.io/minio/openldap
+        image: donatello/openldap
```
Comment (harshavardhana) [inline .github/workflows/iam-integrations.yaml]: anything changed that is needed for this - same commitID

```
4a5b029b01a3e0df4a982a67c19f4b437bc2bb1749cc1e55c769fd2026df2bb2
```

File: .github/workflows/iam-integrations.yaml [hunk 2]
```diff
@@ -61,64 +38,41 @@ jobs:
...
-
-  iam-etcd-test:
-    name: Etcd Backend + LDAP Tests with Go ${{ matrix.go-version }}
-    runs-on: ubuntu-latest
-
-    services:
-      openldap:
-        image: quay.io/minio/openldap
+      openid:
+        image: donatello/dex
```
Comment (harshavardhana) [inline .github/workflows/iam-integrations.yaml]:
```suggestion
        image: quay.io/minio/dex
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: .github/workflows/iam-integrations.yaml (hunk 1 above)
Comment (harshavardhana) [inline .github/workflows/iam-integrations.yaml]:
```suggestion
        image:  quay.io/minio/openldap
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: .github/workflows/iam-integrations.yaml (hunk 1 above)
Comment (donatello) [reply .github/workflows/iam-integrations.yaml]: The ldif file is updated for ldap image

**Review by harshavardhana — APPROVED** [review]

File: .github/workflows/iam-integrations.yaml [hunk 3]
```diff
@@ -12,43 +12,20 @@ concurrency:
...
-  ldap-test:
-    name: LDAP Tests with Go ${{ matrix.go-version }}
+  iam-matrix-test:
+    name: "[Go=${{ matrix.go-version }}|ldap=${{ matrix.ldap }}|etcd=${{ matrix.etcd }}|openid=${{ matrix.openid }}]"
     runs-on: ubuntu-latest
 
     services:
       openldap:
-        image: quay.io/minio/openldap
+        image:  quay.io/minio/openldap
```
Comment (harshavardhana) [inline .github/workflows/iam-integrations.yaml]:
```suggestion
        image: quay.io/minio/openldap
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**